File: Domain/Models/Accounts/Account.cs
using System.ComponentModel.DataAnnotations;
using Domain.Enums;

namespace Domain.Models.Accounts;

/// <summary>
/// Base class for all accounts.
/// </summary>
public class Account
{
    /// <summary>
    /// The account ID
    /// </summary>
    [Required]
    public Guid Id { get; set; }

    /// <summary>
    /// The account name
    /// </summary>
    [Required]
    public required string Name { get; set; }

    /// <summary>
    /// the account balance
    /// </summary>
    [Required]
    public double Balance { get; set; }

    /// <summary>
    /// The <see cref="AccountType"/> of the account
    /// </summary>
    [Required]
    public AccountType AccountType { get; set; }

    /// <summary>
    /// The description of the account
    /// </summary>
    public string? Description { get; set; }

    /// <summary>
    /// The interest of the account
    /// </summary>
    public double? InterestRate { get; set; }

    /// <summary>
    /// The required monthly payment
    /// </summary>
    public double? MonthlyPayment { get; set; }
}

File: client-app/src/models/account.ts
import {Transaction} from "./transaction.ts";

export interface Account {
    id: string;
    name: string;
    balance: number;
    available: number;
    description?: string;
    accountType: string;
    interestRate?: number;
    monthlyPayment?: number;
    transactions: Transaction[];
}

export class Account implements Account {

    constructor(
        id: string,
        name: string,
        balance: number,
        accountType: string,
        available: number,
        description?: string,
        interestRate?: number,
        monthlyPayment?: number,
        transactions?: Transaction[],
    ) {
        this.id = id;
        this.name = name;
        this.balance = balance;
        this.description = description;
        this.available = available;
        this.interestRate = interestRate;
        this.monthlyPayment = monthlyPayment;
        this.accountType = accountType;
        this.transactions = transactions ?? [];
    }
}